llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -2476,9 +2480,10 @@ class VPReplicateRecipe : public VPRecipeWithIRFlags {
 
 public:
   VPReplicateRecipe(Instruction *I, ArrayRef<VPValue *> Operands,
-                    bool IsUniform, VPValue *Mask = nullptr)
+                    bool IsUniform, VPValue *Mask = nullptr,
+                    VPIRMetadata Metadata = {})
       : VPRecipeWithIRFlags(VPDef::VPReplicateSC, Operands, *I),
-        IsUniform(IsUniform), IsPredicated(Mask) {
+        VPIRMetadata(Metadata), IsUniform(IsUniform), IsPredicated(Mask) {
```

Comment:
Should this be
```suggestion
        VPIRMetadata(I, Metadata), IsUniform(IsUniform), IsPredicated(Mask) {
```
to propagate `I`'s metadata when `Metadata` if empty, as in the default?

Reply:
Updated to use `VPIRMetadata(I, LVer)`, to be passed in